src/main/scala/firrtl/FirrtlExecutionOptions.scala

```diff
@@ -0,0 +1,752 @@
+
+  /** Get the user-specified [[OutputConfig]]
+    *
+    * @return the output configuration
+    */
+  def getOutputConfig(): OutputConfig =
+    if (firrtlOptions.emitOneFilePerModule) {
+      OneFilePerModule(firrtlOptions.targetDirName)
+    } else {
+      SingleFile(getBuildFileName(firrtlOptions.outputSuffix, firrtlOptions.outputFileNameOverride))
+    }
+
+  /** Get the user-specified targetFile assuming [[OutputConfig]] is [[SingleFile]]
+    *
+    * @return the targetFile as a String
+    */
+  def getTargetFile(): String = getOutputConfig match {
+    case SingleFile(targetFile) => targetFile
+    case _                      => throw new Exception("OutputConfig is not SingleFile!") }
+
+  parser.note("Common Options")
+
+  parser.opt[String]("top-name")
+    .abbr("tn")
+    .valueName("<top-level-circuit-name>")
+    .action( (x, c) => c :+ TopNameAnnotation(x) )
+    .maxOccurs(1)
+    .text("This options defines the top level circuit, defaults to dut when possible")
+
+  parser.opt[String]("target-dir")
+    .abbr("td")
+    .valueName("<target-directory>")
+    .action( (x, c) => c ++ Seq(TargetDirAnnotation(x), BlackBoxTargetDirAnno(x)) )
+    .maxOccurs(1)
+    .text(s"Work directory for intermediate files/blackboxes, default is ${FirrtlExecutionOptions().targetDirName}")
+
+  parser.opt[String]("log-level")
+    .abbr("ll")
+    .valueName("<Error|Warn|Info|Debug|Trace>")
+    .action( (x, c) => c :+ LogLevelAnnotation(LogLevel(x)) )
+    .validate{ x =>
+      lazy val msg = s"$x bad value must be one of error|warn|info|debug|trace"
+      if (Array("error", "warn", "info", "debug", "trace").contains(x.toLowerCase)) { parser.success      }
+      else                                                                          { parser.failure(msg) }}
+    .maxOccurs(1)
+    .text(s"Sets the verbosity level of logging, default is ${FirrtlExecutionOptions().globalLogLevel}")
+
+  parser.opt[Seq[String]]("class-log-level")
+    .abbr("cll")
+    .valueName("<FullClassName:[Error|Warn|Info|Debug|Trace]>[,...]")
+    .action( (x, c) => c ++ (x.map { y =>
+                val className :: levelName :: _ = y.split(":").toList
+                val level = LogLevel(levelName)
+                ClassLogLevelAnnotation(className, level) }) )
+    .maxOccurs(1)
+    .text(s"This defines per-class verbosity of logging")
+
+  parser.opt[Unit]("log-to-file")
+    .abbr("ltf")
+    .action( (x, c) => c :+ LogToFileAnnotation )
+    .maxOccurs(1)
+    .text(s"default logs to stdout, this flags writes to topName.log or firrtl.log if no topName")
+
+  parser.opt[Unit]("log-class-names")
+    .abbr("lcn")
+    .action( (x, c) => c :+ LogClassNamesAnnotation )
+    .maxOccurs(1)
+    .text(s"shows class names and log level in logging output, useful for target --class-log-level")
+
+  parser.arg[String]("<arg>...")
+    .unbounded()
+    .optional()
+    .action( (x, c) => c :+ ProgramArgsAnnotation(x) )
+    .text("optional unbounded args")
```

Comment:
What are these used for?

Reply:
I'm not sure... they were there, so I didn't delete them (and I did not poke through the logs to try and figure it out). I highly, highly expect that these are some vestigial feature that nobody is using. I can look into it and throw a deprecation warning if they're used. Otherwise, I'm fine to get rid of it and deal with the consequences later...

Reply:
These are used by the Chisel Tutorial so that command line execution scripts can pass program arguments to the various main functions in the project. A user might similarly want to use this function to write bash scripts that invoke mains.

Reply:
Thanks for the necessary clarification! "Used by the tutorial" is about as far as possible from "vestigial feature"...